pkg/vault/error.go

```diff
@@ -0,0 +1,60 @@
+package vault
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+	"os"
+	"strings"
+	"syscall"
+)
+
+// Error is the custom error type for this package
+type Error struct {
```

Comment:
Feel like this should be named something more specific, to easy to confuse error and Error and mismatch them.  I would say something like StimError, or even StimVaultError